agent/api/container/status/managedagentstatus_test.go

```diff
@@ -0,0 +1,95 @@
+// +build unit
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package status
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTerminal(t *testing.T) {
+	terminalStatus := ManagedAgentStopped
+	runningStatus := ManagedAgentRunning
+	assert.True(t, terminalStatus.Terminal())
+	assert.False(t, runningStatus.Terminal())
+}
+
+func TestIsRunning(t *testing.T) {
+	terminalStatus := ManagedAgentStopped
+	runningStatus := ManagedAgentRunning
+	assert.True(t, runningStatus.IsRunning())
+	assert.False(t, terminalStatus.IsRunning())
+}
+
+func TestShouldReportManagedAgentStatusToBackend(t *testing.T) {
+	// We should not report ManagedAgentStatusNone
+	var managedAgentStatus ManagedAgentStatus
+	assert.Equal(t, false, managedAgentStatus.ShouldReportToBackend())
+	// We should report ManagedAgentCreated (ie PENDING)
+	managedAgentStatus = ManagedAgentCreated
+	assert.Equal(t, true, managedAgentStatus.ShouldReportToBackend())
+	// We should report ManagedAgentRunning (ie RUNNING)
+	managedAgentStatus = ManagedAgentRunning
+	assert.Equal(t, true, managedAgentStatus.ShouldReportToBackend())
+	// We should report ManagedAgentStooped (ie STOPPED)
+	managedAgentStatus = ManagedAgentStopped
+	assert.Equal(t, true, managedAgentStatus.ShouldReportToBackend())
+}
+
+func TestManagedAgentBackendStatus(t *testing.T) {
+	// BackendStatus is "PENDING" when managedAgent status is ManagedAgentStatusNone
+	var managedAgentStatus ManagedAgentStatus
+	assert.Equal(t, managedAgentStatus.BackendStatus(), "PENDING")
```

Comment:
nit: expected is first: `assert.Equal(t, "PENDING", managedAgentStatus.BackendStatus())`

Reply:
updated